1-mandatory/1-alarmclock/alarmclock.js

```diff
@@ -1,4 +1,49 @@
-function setAlarm() {}
+function setAlarm() {
+  let second = document.querySelector("#alarmSet").value;
+  let timeRemaining = document.querySelector("#timeRemaining");
+  let setButton = document.querySelector("#set");
+  const flash = ["red", "white"];
+
+  let minute = parseInt(second / 60);
+  second %= 60;
+
+  const stopTimeRemaining = setInterval(() => {
+
+    if (second === -1 && minute > 0) {
+      minute--;
+      second = 59;
+    }
+
+    if (second < 10)
+      second = second.toLocaleString('en-US', {
+        minimumIntegerDigits: 2,
+        useGrouping: false
+      })
+
+    if (minute < 10)
+      minute = minute.toLocaleString('en-US', {
+        minimumIntegerDigits: 2,
+        useGrouping: false
+      })
+
+    if (second >= 0) {
+      timeRemaining.innerText = `Time Remaining: ${minute}:${second}`;
+      second--;
+    }
+
+    if (second === -1 && minute === "00") {
```

Comment:
This works, though generally conditionals should be checking logically relevant conditions - what you care about here is when the overall time reaches `0`, so checking that a string equals some string and that seconds is negative makes this a little hard to understand. 

I'd recommend converting the entered time into seconds and keeping track of that as your method of tracking time. This means you can use existing JS libraries to format things for you (like the `Date` object`:
```
const formattedTime  = new Date(seconds * 1000).toISOString().slice(11, 19);
// outputs string like `"hh:mm:ss"` e.g. `"00:09:12"` for 9min12sec
```

you can then make this (seemingly random line of code) make sense by giving it a context-appropriate function name/comment:
```
// formats time in "hh:mm:ss" format
const formatTime = (seconds) => {
   return new Date(seconds * 1000).toISOString().slice(11, 19);
}
```